### PR TITLE
Better error messages

### LINKED ISSUE #23: "flash" errors
not in the sense of _adobe flash_ but a way to add error/validation messages to a page (like name taken or login incorrect).

Right now we just render an error page with a back button.

### LINKED ISSUE #66: localized error pages
tracking issue for all the `// TODO: localized error pages`s everywhere.

To tackle this `go.mindeco.de/http/render` needs to be patched so that [the Error handler](https://pkg.go.dev/go.mindeco.de/http/render#Renderer.Error) can be overwritten.

Then we receive the errors directly and can pick localizations for specific errors.

### PR DESCRIPTION
**Pull request by cryptix**
fixes #66 
fixes #23 

Most importantly: I added `errhandler.go` and `flashes.go` to `web/errors`, these use `localizeError()` which translates a few hardcoded error messages to the users locale. I updated the english defaults accordingly.

Most of the handlers now have `flashes weberrors.FlashHelper` which has three methods:

* `AddMessage(rw http.ResponseWriter, req *http.Request, label string)`
* `AddError(rw http.ResponseWriter, req *http.Request, err error)`
* `GetAll(rw http.ResponseWriter, req *http.Request) ([]FlashMessage, error)`

To render these there is a new base template `flashes.tmpl` which can be used with `{{ tempalte "flashes" .}}`, See the overviews of admin/members, admin/aliases, etc.

Noteworthy:
*  `go.mindeco.de/http/tester@v1.11.0` now handles cookies automatically which made it possible to simplify quite a bit of the testing code that uses login and csrf.

TODO:

* [x] style the flash messages (right now they are just red and green text)
* [x] flashify login errors
* [x] flashify notice edits

### CODE REVIEW

File: web/templates/error.tmpl [hunk 1]
```diff
@@ -7,11 +7,7 @@
   <div class="col-sm-12">
     <p id="errBody">{{.Err}}</p>
     <p>
-      <!-- this violates the CSP, since it's an inline script.
-      We probably want to solve https://github.com/ssb-ngi-pointer/go-ssb-room/issues/23
-      but until then it might be nice to include a _previous_ data from the rendere with an url of where to go.
-      -->
-      <a href="javascript:history.back()" class="btn btn-primary">Back</a>
+      <a href="{{.BackURL}}" class="btn btn-primary">Back</a>
```
Comment (staltz) [inline web/templates/error.tmpl:10]: Yay!

**Review by staltz — APPROVED** [review]
The Go looks good, and I took a look at the styles, they seem simple and obvious, I don't think we need to improve the looks (an error anyway looks unpleasant, by design). What's important is that it's a consistent style, and that it clearly communicates an error.

**Comment by staltz** [comment]
Oh, and can other pages use the `BackURL` in the templates? I mean this as a question of possibility, not as something to be done within this PR.

**Comment by cryptix** [comment]
> can other pages use the BackURL in the templates?

Not yet, right now it's handled in the errorhandler, which also looks at the status code to to a kinda-sane thing:

https://github.com/ssb-ngi-pointer/go-ssb-room/pull/115/files#diff-f49e5d91dc7f3e5f05f0b6b833a9b0a7b306612fae102bae8f7b95bb46170e86R55-R64

Where would you want to use it?

**Comment by cblgh** [comment]
🎉

**Comment by staltz** [comment]
> Where would you want to use it?

Remove member confirmation, revoke invite confirmation, maybe other places.

**Pull request merged**